openstack/sharedfilesystems/v2/shares/requests.go

```diff
@@ -66,6 +69,81 @@ func Create(client *gophercloud.ServiceClient, opts CreateOptsBuilder) (r Create
 	return
 }
 
+// ListOpts holds options for listing Shares. It is passed to the
+// shares.List function.
+type ListOpts struct {
```

Comment:
As best as I can tell, this looks correct. I've found some fields also defined in:

* https://github.com/openstack/manila/blob/e4acaa5bf14c1e188a21aab5f15d9613ebb0aa6e/manila/api/v1/shares.py#L123
* https://github.com/openstack/manila/blob/e4acaa5bf14c1e188a21aab5f15d9613ebb0aa6e/manila/api/v1/shares.py#L187

The v1 code is called from v2 by way of inheritance: https://github.com/openstack/manila/blob/e4acaa5bf14c1e188a21aab5f15d9613ebb0aa6e/manila/api/v2/shares.py#L421-L450

Reply:
I've added `metadata`, `extra_specs` and the missing fields from your second link. Thanks!

Reply:
Nice - thank you.